### PR TITLE
[WIP] feat: add wds version

### PR DESCRIPTION
**Pull request by sayakpaul**
PR is not complete yet. 

@zhuole1025 could you check if the current implementation works for you?

### CODE REVIEW

File: train_flux/config.yaml [hunk 1]
```diff
@@ -23,25 +23,7 @@ train:
...
-      ],
-      "rule": [
-        "/mnt/petrelfs/zhuole/data/metadata_clean/geneval_pairs_train.json", # 129321
-        "/mnt/petrelfs/zhuole/data/metadata_clean/t2i_pairs_train.json" # 55388
-      ],
-      "editing": [
-        "/mnt/petrelfs/zhuole/data/metadata_clean/editing_pairs_train.json" # 616409
-      ]
-    }
+    path: "pipe:curl -s -f -L https://huggingface.co/datasets/diffusion-cot/GenRef-wds/resolve/main/genref_{0..2}.tar"
```
Comment (sayakpaul) [inline train_flux/config.yaml]: `webdataset` related change. For the training run, I just used three shards.

File: train_flux/train/data_wds.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
+import math
```
Comment (sayakpaul) [inline train_flux/train/data_wds.py]: Can rename to `data.py` and remove the existing `data.py` file once the changes are finalized.

File: train_flux/train/callbacks.py [hunk 3]
```diff
@@ -44,7 +44,7 @@ def on_train_batch_end(self, trainer, pl_module, outputs, batch, batch_idx):
         self.total_steps += 1
         
         # Update split ratios
-        trainer.train_dataloader.dataset._update_split_ratios(self.total_steps)
+        trainer.train_dataloader.dataset._update_split_ratios()
```
Comment (sayakpaul) [inline train_flux/train/callbacks.py:47]: No need to pass `total_steps` for this update when using `webdataset`.

File: train_flux/train/train.py [hunk 4]
```diff
@@ -75,32 +74,32 @@ def main():
...
+        # val_dataset = ImageConditionDataset(
+        #     training_config["dataset"]["val_path"],
+        #     condition_size=training_config["dataset"]["condition_size"],
+        #     target_size=training_config["dataset"]["target_size"],
+        #     condition_type=training_config["condition_type"],   
+        #     root_dir=training_config["dataset"]["val_root_dir"],
+        #     drop_text_prob=0,
+        #     drop_image_prob=0,
+        #     drop_reflection_prob=0,
+        # )
```
Comment (sayakpaul) [inline train_flux/train/train.py]: Temporary change as I don't have access to the validation split.

File: train_flux/train/train.py [hunk 5]
```diff
@@ -75,32 +74,32 @@ def main():
...
     else:
         raise NotImplementedError
 
-    print("Dataset length:", len(dataset))
+    # print("Dataset length:", len(dataset))
     train_loader = DataLoader(
         dataset,
         batch_size=training_config["batch_size"],
-        shuffle=True,
+        # shuffle=True,
```
Comment (sayakpaul) [inline train_flux/train/train.py]: We're already shuffling inside `webdataset` so no need.
